ingestion/src/metadata/ingestion/source/database/redshift/queries.py

```diff
@@ -229,3 +229,48 @@
 
 
 REDSHIFT_TEST_PARTITION_DETAILS = "select * from SVV_TABLE_INFO limit 1"
+
+
+# Redshift views definitions only contains the select query
+# hence we are appending "create view <schema>.<table> as " to select query
+# to generate the column level lineage
+REDSHIFT_GET_ALL_RELATIONS = """
+    SELECT
+        c.relkind,
+        n.oid as "schema_oid",
+        n.nspname as "schema",
+        c.oid as "rel_oid",
+        c.relname,
+        CASE c.reldiststyle
+        WHEN 0 THEN 'EVEN' WHEN 1 THEN 'KEY' WHEN 8 THEN 'ALL' END
+        AS "diststyle",
+        c.relowner AS "owner_id",
+        u.usename AS "owner_name",
+        TRIM(TRAILING ';' FROM 
+        'create view ' || n.nspname || '.' || c.relname || ' as ' ||pg_catalog.pg_get_viewdef(c.oid, true))
+        AS "view_definition",
+        pg_catalog.array_to_string(c.relacl, '\n') AS "privileges"
+    FROM pg_catalog.pg_class c
+            LEFT JOIN pg_catalog.pg_namespace n ON n.oid = c.relnamespace
+            JOIN pg_catalog.pg_user u ON u.usesysid = c.relowner
+    WHERE c.relkind IN ('r', 'v', 'm', 'S', 'f')
+        AND n.nspname !~ '^pg_' {schema_clause} {table_clause}
+    UNION
+    SELECT
+        'r' AS "relkind",
+        s.esoid AS "schema_oid",
+        s.schemaname AS "schema",
+        null AS "rel_oid",
+        t.tablename AS "relname",
+        null AS "diststyle",
+        s.esowner AS "owner_id",
+        u.usename AS "owner_name",
+        null AS "view_definition",
+        null AS "privileges"
+    FROM
+        svv_external_tables t
+        JOIN svv_external_schemas s ON s.schemaname = t.schemaname
+        JOIN pg_catalog.pg_user u ON u.usesysid = s.esowner
+    where 1 {schema_clause} {table_clause}
```

Comment:
We should avoid formatting queries this way - this is vulnerable to SQL injection. Instead, we should parametrize the queries -- see the screenshot for an example. There might be a low probability a user will actually be able to inject any kind of SQL we should still have this air tight.

https://github.com/open-metadata/OpenMetadata/assets/13626425/41be5ec4-2858-4b91-a372-a295678a4875

Reply:
@TeddyCr regarding this we actually want to inject the sql in the query. [this](https://github.com/open-metadata/OpenMetadata/blob/e8c588e1f79d738fe3e5669ec8f54ead8bd25112/ingestion/src/metadata/ingestion/source/database/redshift/utils.py#L380) is where the query is being called. we are adding the condition of table and schema only when we have the information available otherwise we fetch all the tables.

Reply:
so I actually tried the same thing and it gets translated  to this `where 1 'schema = ''public''' ''` where the condition is within quotes and throws a syntax error

Reply:
Oh my bad, did not see we were passing an identifier. Make sense. We cannot parametrize identifier from what I understand.